getting_started/scripting/gdscript/gdscript_basics.rst

```diff
@@ -1463,63 +1463,114 @@ freed.
+   func _ready():
+      var notifier = your_notifier.new()
+      var handler = your_handler.new()
 
-    func _at_some_func():
-        instance.connect("your_signal_name", self, "_callback_no_args")
-        instance.connect("your_signal_name_with_args", self, "_callback_args")
+      notifier.connect("your_notify", handler, "your_handler")
 
-It is also possible to bind arguments to a signal that lacks them with
-your custom values:
+GDScript's signals also have the capability to bind arguments to connections
+between a signal and a slot. This means that when the signal is emitted,
+calling the connected slot, the bound argument is always given to the slot.
+These bound arguments are a property of the connection rather than the signal
+or the slot, which allows each connection to have unique bindings.
 
+An example that uses this feature:
 ::
+   
+   func pressed_handler(which):
+      print("A button was pressed! Button's name was:", which.get_name())
 
-    func _at_some_func():
-        instance.connect("your_signal_name", self, "_callback_args", [22, "hello"])
+   func _ready():
+      for button in get_node("buttons").get_children()
+         # Connect the button's 'pressed' signal to our 'pressed_handler' slot
+         # Bind the button to the connection so we know which button was pressed
+         button.connect("pressed", self, "pressed_handler", [button])
 
-This is useful when a signal from many objects is connected to a
-single callback and the sender must be identified:
+To bring all of this together is the `emit_signal` keyword which triggers the
+broadcast of the signal, along with all the arguments, to the various slots
+that are connected to that signal.
+
+Here is how we would extend one of the previous examples to use all of the features
+GDScript signals have:
 
 ::
 
-    func _button_pressed(which):
-        print("Button was pressed: ", which.get_name())
+   your_notifier.gd
+   ----------------
 
-    func _ready():
-        for b in get_node("buttons").get_children():
-            b.connect("pressed", self, "_button_pressed",[b])
+   signal your_notify(data)
+
+   var your_data = 42
+
+   func _process(delta):
+      if delta == your_data:
+         emit_signal("your_notify", data)
+
+::
+
+   your_handler.gd
+   ---------------
 
-Finally, emitting a custom signal is done by using the
-Object.emit_signal method:
+   func your_handler(data, obj):
+      print("Your handler was called from: ", obj.get_name(), with data: ", data)
 
 ::
 
-    func _at_some_func():
-        emit_signal("your_signal_name")
-        emit_signal("your_signal_name_with_args", 55, 128)
-        some_instance.emit_signal("some_signal")
+   your_game.gd
+   ------------
+
+   func _ready():
+      var notifier = your_notifier.new()
+      var handler = your_handler.new()
+
+      notifier.connect("your_notify", handler, "your_handler", [notifier])
+
+Note that bound arguments are added to the end of the arguments sent to
+the slot.
```

Comment:
What is this saying exactly? It's not clear. Are you referring to default arguments of built-in functions, or declared arguments of user-defined functions?

Reply:
I was referring to the behavior that bound arguments (the fourth parameter of the `connect` call) are appended to the argument list defined for the signal rather than put in front of the argument list when calling the connected method. Perhaps that information belongs in the documentation of `Object.connect` rather than here?